packages/astro/src/runtime/server/endpoint.ts

```diff
@@ -49,6 +51,11 @@ export async function renderEndpoint(
 	}
 
 	const response = await handler.call(mod, context);
+
+	if (!response || response instanceof Response === false) {
+		throw new AstroError(EndpointDidNotReturnAResponse)
+	}
```

Comment:
Isn't this handled by the `renderPage` function of the pipeline? If not, then a test should be cover this edge case

Reply:
The `renderPage` function wouldn't run for endpoints. I'm guessing you meant `RenderContext.render()` - it checks the return of the middleware. Failures from the absence of a response occur before control flow returns to it.

An error here is not new, it was just thrown by accessing `response.status` before, now it thrown explicitly. A test would work the same before and after this PR.

Reply:
A different error would be logged, but the tests will have access to a 500 response either way.

Reply:
Still, I believe it's worth expanding our test suite in this case. Any `throw` could return a 500, although this is a specific error that Astro decides to handle a meaningful way with a proper message.

Such cases should to be tested. We do this in our middleware, I don't see any reason why we shouldn't do the same here.

Reply:
Okay, that makes sense. I had thought you meant a regression test.